File: java/test/in3/ipfs/ApiTest.java
package in3.ipfs;

import in3.Chain;
import in3.IN3;
import in3.IN3MockBuilder;
import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

public class ApiTest {

  private IN3 in3;

  @BeforeEach
  public void setupClient() {
    String[][] mockedResponses = {
        {"ipfs_get", "ipfs_get.json"},
        {"ipfs_put", "ipfs_put.json"}};
    IN3MockBuilder builder = new IN3MockBuilder(Chain.IPFS);
    in3                    = builder.constructClient(mockedResponses);
  }

  @Test
  public void put_success() {
    String content   = "Lorem ipsum dolor sit amet";
    String multihash = in3.getIpfs().put(content);
    Assertions.assertNotNull(multihash);
    Assertions.assertEquals("QmbGySCLuGxu2GxVLYWeqJW9XeyjGFvpoZAhGhXDGEUQu8", multihash);
  }

  @Test
  public void put_failure() {
    String nullable = null;
    Assertions.assertThrows(IllegalArgumentException.class, () -> {
      in3.getIpfs().put(nullable);
    });
  }

  @Test
  public void get_success() {
    String multihash = "QmbGySCLuGxu2GxVLYWeqJW9XeyjGFvpoZAhGhXDGEUQu8";
    byte[] content   = in3.getIpfs().get(multihash);
    Assertions.assertNotNull(content);
    Assertions.assertEquals("Lorem ipsum dolor sit amet", new String(content));
  }

  @Test
  public void get_failure() {
    String multihash = null;
    Assertions.assertThrows(IllegalArgumentException.class, () -> {
      in3.getIpfs().get(multihash);
    });
  }
}

File: dotnet/Test/Ipfs/ApiTest.cs
using System;
using System.Text;
using System.Threading.Tasks;
using In3;
using NUnit.Framework;

namespace Test.Ipfs
{
    public class ApiTest
    {
        private IN3 _in3;

        [SetUp]
        public void Setup()
        {
            string[][] mockedResponses = {
                new[] {"ipfs_get", "ipfs_get.json"},
                new[] {"ipfs_put", "ipfs_put.json"}
            };

            ClientBuilder builder = new ClientBuilder(Chain.Ipfs); ;
            _in3 = builder.ConstructClient(mockedResponses);
        }

        [Test]
        public async Task Put_Success()
        {
            string content = "Lorem ipsum dolor sit amet";

            string multihash = await _in3.Ipfs.Put(content);

            Assert.NotNull(multihash);
            Assert.That(multihash, Is.EqualTo("QmbGySCLuGxu2GxVLYWeqJW9XeyjGFvpoZAhGhXDGEUQu8"));
        }

        [Test]
        public void Put_Failure()
        {
            string nullString = null;

            Assert.ThrowsAsync<ArgumentException>(async () => await _in3.Ipfs.Put(nullString));
        }

        [Test]
        public async Task Get_Success()
        {
            string multihash = "QmbGySCLuGxu2GxVLYWeqJW9XeyjGFvpoZAhGhXDGEUQu8";

            byte[] content = await _in3.Ipfs.Get(multihash);

            Assert.NotNull(content);
            Assert.That(content, Is.EqualTo(Encoding.UTF8.GetBytes("Lorem ipsum dolor sit amet")));
        }

        [Test]
        public void Get_Failure()
        {
            string nullString = null;

            Assert.ThrowsAsync<ArgumentException>(async () => await _in3.Ipfs.Get(nullString));
        }
    }
}